src/components/ModalWindow/ModalWindow.jsx

```diff
@@ -1,30 +1,44 @@
-import {Form} from "../Form/Form.jsx"
-import {Modal, Box} from '@mui/material'
 import propTypes from "prop-types";
-import styles from "./ModalWindow.module.css"
+import { Form } from "../Form/Form.jsx";
+import { Modal, Box } from "@mui/material";
+import useModalClose from "./hooks/useModalClose.js";
+import styles from "./ModalWindow.module.css";
+import { OperationButton } from "../OperationButton/OperationButton.jsx";
 
+export const ModalWindow = ({
+  isOpen,
+  setIsModalOpen,
+  infoItem,
+  flagEdit,
+  onEdit,
+}) => {
+  const { handleClose, handleCloseBtn } = useModalClose({ setIsModalOpen });
 
-
-export const ModalWindow=({onOpen, onClose, infoItem, flagEdit, onEdit})=>{
-  
-    return (
-      <Modal open={onOpen} onClose={onClose}>
-        <Box className={styles.module}>
-          <Form
-            flagEdit={flagEdit}
-            defaultInputForm={infoItem}
-            textButton="Edit element"
-            onEdit={onEdit}
-          />
-        </Box>
-      </Modal>
-    );
-}
+  return (
+    <Modal open={isOpen} onClose={handleClose}>
+      <Box className={styles.module}>
+        <div className={styles.btnClose} onClick={handleCloseBtn}>
```

Comment:
I added onClick on the div because when I tested it in mobile version some users may have problems with accessing the close button due to the small screen, so I make the area around the button also sensitive for close action.